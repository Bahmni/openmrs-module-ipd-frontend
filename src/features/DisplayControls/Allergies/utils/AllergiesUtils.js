import axios from "axios";
import { ALLERGIES_BASE_URL } from "../../../../constants";

export const fetchAllergiesData = async (patientUuid) => {
  const FETCH_ALLERGIES_URL = `${ALLERGIES_BASE_URL}?patient=${patientUuid}&_summary=data`;
  try {
    const response = await axios.get(FETCH_ALLERGIES_URL);
    return response;
  } catch (error) {
    return error;
  }
};
