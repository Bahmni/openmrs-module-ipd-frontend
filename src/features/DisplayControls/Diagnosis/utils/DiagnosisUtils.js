import axios from "axios";
import { DIAGNOSIS_SEARCH_URL } from "../../../../constants";

export const getPatientDiagnosis = async (patientUuid) => {
  try {
    const response = await axios.get(DIAGNOSIS_SEARCH_URL, {
      params: { patientUuid: patientUuid },
      withCredentials: true,
    });
    return response.data;
  } catch (error) {
    return error;
  }
};

export const diagnosisHeaders = [
  {
    id: "1",
    header: "Diagnosis",
    key: "diagnosis",
    isSortable: false,
  },
  {
    id: "2",
    header: "Order",
    key: "order",
    isSortable: true,
  },
  {
    id: "3",
    header: "Certainty",
    key: "certainty",
    isSortable: true,
  },
  {
    id: "4",
    header: "Status",
    key: "status",
    isSortable: true,
  },
  {
    id: "5",
    header: "Diagnosed By",
    key: "diagnosedBy",
    isSortable: false,
  },
  {
    id: "6",
    header: "Diagnosis Date ",
    key: "diagnosisDate",
    isSortable: true,
  },
];
