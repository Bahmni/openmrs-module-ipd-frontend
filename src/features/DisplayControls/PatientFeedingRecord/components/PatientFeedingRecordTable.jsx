import React from "react";
import {
  DataTable,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from "carbon-components-react";

const PatientFeedingRecordTable = (props) => {
  const { rows, headers, useZebraStyles } = props;

  return (
    <DataTable rows={rows} headers={headers} useZebraStyles={useZebraStyles}>
      {({ rows, headers, getTableProps, getHeaderProps, getRowProps }) => (
        <Table {...getTableProps()} useZebraStyles>
          <TableHead>
            <TableRow>
              {headers.map((header, index) => (
                <TableHeader
                  key={index + header.key}
                  {...getHeaderProps({ header })}
                >
                  {header.header}
                </TableHeader>
              ))}
            </TableRow>
          </TableHead>
          <TableBody>
            {rows.map((row, index) => (
              <TableRow
                key={index + row.id}
                {...getRowProps({ row })}
                data-testid="table-body-row"
              >
                {row.cells.map((cell) => (
                  <TableCell key={cell.id}>{cell.value}</TableCell>
                ))}
              </TableRow>
            ))}
          </TableBody>
        </Table>
      )}
    </DataTable>
  );
};

export default PatientFeedingRecordTable;
