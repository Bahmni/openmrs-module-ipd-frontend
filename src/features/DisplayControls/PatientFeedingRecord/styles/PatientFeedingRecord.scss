.no-pfr-data {
  padding-top: 10px;
}