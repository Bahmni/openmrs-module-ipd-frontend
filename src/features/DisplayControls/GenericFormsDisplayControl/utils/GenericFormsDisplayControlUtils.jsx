import axios from "axios";
import { FETCH_FORM_DETAILS_URL } from "../../../../constants";
import { getFetchFormTranslationsUrl } from "../../../../utils/CommonUtils";

export const fetchAllConceptsForForm = async (formUuid) => {
  try {
    const response = await axios.get(
      FETCH_FORM_DETAILS_URL.replace("{formUuid}", formUuid),
      {
        withCredentials: true,
      }
    );
    return response.data;
  } catch (e) {
    console.error(e);
  }
};

export const fetchFormTranslations = async (formName, formUuid) => {
  try {
    const response = await axios.get(
      getFetchFormTranslationsUrl(formName.replace(" ", "+"), formUuid),
      {
        withCredentials: true,
      }
    );
    return response.data;
  } catch (e) {
    console.error(e);
  }
};
