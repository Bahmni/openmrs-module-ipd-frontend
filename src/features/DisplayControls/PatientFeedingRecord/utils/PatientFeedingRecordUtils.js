import React from "react";
import {
  defaultDateTimeFormat,
} from "../../../../constants";
import { formatDate } from "../../../../utils/DateTimeUtils";
import { FormattedMessage } from "react-intl";

export const patientFeedingRecordHeaders = [
  {
    key: "dateAndTime",
    header: "Date and Time",
  },
  {
    key: "shift",
    header: "Shift",
  },
  {
    key: "route",
    header: "Route",
  },
  {
    key: "feedType",
    header: "Feed Type",
  },
  {
    key: "amount",
    header: "Amount",
  },
];

export const transformObsData = (obsData, config) => {
  const transformedData = [];
  let quantityUnits = "";
  obsData.forEach((obs) => {
    const dateTime = obs.groupMembers.find((member) =>
      config?.timeConceptNames?.includes(member?.concept?.name)
    );
    const shiftValue = obs.groupMembers.find(
      (member) => member?.concept?.name === config?.shiftValueConcept
    );
    const routeValue = obs.groupMembers.find(
      (member) => member?.concept?.name === config?.routeValueConcept
    );
    const feedTypeValue = obs.groupMembers.find(
      (member) => member?.concept?.name === config?.feedTypeConcept
    );
    const amountValue = obs.groupMembers.find(
      (member) => member?.concept?.name === config?.amountValueConcept
    );
    quantityUnits = amountValue?.concept?.units;
    transformedData.push({
      id: obs.uuid,
      dateAndTime: formatDate(dateTime?.value, defaultDateTimeFormat),
      shift: shiftValue && shiftValue.valueAsString,
      route: routeValue && routeValue.valueAsString,
      feedType: feedTypeValue && feedTypeValue.valueAsString,
      amount: amountValue && amountValue.value + " " + quantityUnits,
    });
  });
  return transformedData;
};

export const NoDataForSelectedPeriod = (
  <FormattedMessage
    id={"NO_PFR_DATA_MESSAGE"}
    defaultMessage={
      "No patient feeding record data is available for the patient in this period"
    }
  />
);