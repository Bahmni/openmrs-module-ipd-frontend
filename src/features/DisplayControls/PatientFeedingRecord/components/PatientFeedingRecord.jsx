import React, { useContext, useEffect, useState } from "react";
import { FormattedMessage } from "react-intl";
import moment from "moment";
import { Button, DataTableSkeleton } from "carbon-components-react";
import { ChevronLeft16, ChevronRight16, Time16 } from "@carbon/icons-react";
import { IPDContext } from "../../../../context/IPDContext";
import {
  displayPeriodTimingsFormat,
  timeFormatFor12Hr,
} from "../../../../constants";
import WarningIcon from "../../../../icons/warning.svg";
import { formatDate } from "../../../../utils/DateTimeUtils";
import {
  currentPeriodRange,
  NotCurrentPeriodMessage,
  isCurrentPeriod,
  getSortedObsData,
  filterDataForRange,
  fetchFormData,
} from "../../IntakeOutput/utils/IntakeOutputUtils";
import {
  patientFeedingRecordHeaders,
  transformObsData,
  NoDataForSelectedPeriod,
} from "../utils/PatientFeedingRecordUtils.js";
import PatientFeedingRecordTable from "./PatientFeedingRecordTable";
import "../styles/PatientFeedingRecord.scss";

const PatientFeedingRecord = () => {
  const { config, isReadMode, visitSummary, patient } = useContext(IPDContext);
  const { patientFeedingRecordConfig = {} } = config;
  const [consolidatedPFRData, setConsolidatedPFRData] = useState({});
  const [isLoading, setIsLoading] = useState(true);
  const [notCurrentPeriod, setNotCurrentPeriod] = useState(false);
  const [patientFeedingRecordData, setPatientFeedingRecordData] = useState([]);
  const { enable24HourTime = {} } = config;
  const {
    periodDetails: periodConfig = {},
    dashboardConfig: dashboardConfig = {},
  } = patientFeedingRecordConfig;
  const currentPeriod = currentPeriodRange(
    isReadMode ? new Date(visitSummary.stopDateTime) : new Date(),
    periodConfig
  );

  const [startEndDates, updatedStartEndDates] = useState({
    startDate: isReadMode ? new Date(visitSummary.stopDateTime) : new Date(),
    endDate: isReadMode ? new Date(visitSummary.stopDateTime) : new Date(),
  });

  const [periodButtonsDisabled, setPeriodButtonsDisabled] = useState({
    previous: false,
    next: isReadMode ? true : false,
  });
  const visitStartDateTime = visitSummary.startDateTime;

  const callFetchPatientFeedingRecordData = async () => {
    try {
      const response = await fetchFormData(
        patient.uuid,
        dashboardConfig.conceptNames,
        dashboardConfig.numberOfVisits
      );
      setPatientFeedingRecordData(response);
    } catch (e) {
      return e;
    } finally {
      setIsLoading(false);
    }
  };
  const handleCurrent = () => {
    setNotCurrentPeriod(false);
    setIsLoading(true);
    updatedStartEndDates({
      startDate: currentPeriod.startDateTime,
      endDate: currentPeriod.endDateTime,
    });
  };

  const handlePrevious = () => {
    setNotCurrentPeriod(true);
    setIsLoading(true);
    updatedStartEndDates({
      startDate: moment(startEndDates.startDate)
        .clone()
        .subtract(periodConfig.durationInHours, "hours"),
      endDate: startEndDates.startDate,
    });
  };

  const handleNext = () => {
    setNotCurrentPeriod(true);
    setIsLoading(true);
    updatedStartEndDates({
      startDate: startEndDates.endDate,
      endDate: moment(startEndDates.endDate)
        .clone()
        .add(periodConfig.durationInHours, "hours"),
    });
  };

  const periodTiming = () => {
    let periodStartDateTime = formatDate(
      startEndDates.startDate,
      displayPeriodTimingsFormat
    );
    let periodEndDateTime = formatDate(
      startEndDates.endDate,
      displayPeriodTimingsFormat
    );
    const [periodStartDate, periodStartTime] = periodStartDateTime.split(" | ");
    const [periodEndDate, periodEndTime] = periodEndDateTime.split(" | ");

    const formattedPeriodStartTime = enable24HourTime
      ? periodStartTime
      : formatDate(startEndDates.startDate, timeFormatFor12Hr);

    const formattedPeriodEndTime = enable24HourTime
      ? periodEndTime
      : formatDate(startEndDates.endDate, timeFormatFor12Hr);

    if (periodStartDate === periodEndDate) {
      return (
        <div className="period-timing">
          {notCurrentPeriod && (
            <div className="not-current-period-message-div">
              <WarningIcon />
              <span className="not-current-period-message">
                {NotCurrentPeriodMessage}
              </span>
            </div>
          )}
          <div className="period-time">
            {periodStartDate} <span className="mr-5">|</span> <Time16 />{" "}
            {formattedPeriodStartTime} <span className="to-text">to</span>{" "}
            <Time16 /> {formattedPeriodEndTime}
          </div>
        </div>
      );
    } else {
      return (
        <div className="period-timing">
          {notCurrentPeriod && (
            <div className="not-current-period-message-div">
              <WarningIcon />
              <span className="not-current-period-message">
                {NotCurrentPeriodMessage}
              </span>
            </div>
          )}
          <div className="period-time">
            {periodStartDate} <span className="mr-5">|</span> <Time16 />{" "}
            {formattedPeriodStartTime} <span className="to-text">to</span>{" "}
            {periodEndDate} <span className="mr-5">|</span> <Time16 />{" "}
            {formattedPeriodEndTime}
          </div>
        </div>
      );
    }
  };

  useEffect(() => {
    isCurrentPeriod(startEndDates, periodConfig)
      ? setNotCurrentPeriod(false)
      : setNotCurrentPeriod(true);

    const filteredData = filterDataForRange(
      patientFeedingRecordData,
      patientFeedingRecordConfig.timeConceptNames,
      startEndDates,
      visitSummary.startDateTime
    );

    const sortedObsData = getSortedObsData(
      filteredData,
      patientFeedingRecordConfig.timeConceptNames
    );

    setPeriodButtonsDisabled({
      previous:
        (isReadMode && patientFeedingRecordData.length === 0) ||
        moment(visitStartDateTime).isBetween(
          startEndDates.startDate,
          startEndDates.endDate,
          null,
          "[)"
        ),
      next: moment(visitSummary.stopDateTime).isBefore(startEndDates.endDate),
    });
    setConsolidatedPFRData(
      transformObsData(sortedObsData, patientFeedingRecordConfig)
    );
    setIsLoading(false);
  }, [startEndDates, patientFeedingRecordData, visitStartDateTime]);

  useEffect(() => {
    updatedStartEndDates({
      startDate: currentPeriod.startDateTime,
      endDate: currentPeriod.endDateTime,
    });
    callFetchPatientFeedingRecordData();
  }, []);
  return (
    <div className="form-dc-container display-container">
      <div className={"navigation"}>
        <div className="period-header">
          <Button
            kind="tertiary"
            isExpressive
            size="small"
            onClick={handleCurrent}
            className="margin-right-10"
            data-testid="current-period"
            disabled={isReadMode}
          >
            <FormattedMessage
              id={"CURRENT_PERIOD"}
              defaultMessage={"Current Period"}
            />
          </Button>
          <Button
            disabled={periodButtonsDisabled.previous}
            renderIcon={ChevronLeft16}
            kind="tertiary"
            isExpressive
            hasIconOnly
            size="sm"
            onClick={handlePrevious}
            className="margin-right-6"
            data-testid="previous-shift"
          />
          <Button
            disabled={periodButtonsDisabled.next}
            renderIcon={ChevronRight16}
            kind="tertiary"
            isExpressive
            hasIconOnly
            size="sm"
            onClick={handleNext}
            className="margin-right-10"
            data-testid="next-shift"
          />
          {periodTiming()}
        </div>
      </div>
      <div>
        {isLoading ? (
          <DataTableSkeleton
            data-testid="datatable-skeleton"
            headers={patientFeedingRecordHeaders}
            aria-label="sample table"
          />
        ) : consolidatedPFRData && consolidatedPFRData.length === 0 ? (
          <div className="no-pfr-data">{NoDataForSelectedPeriod}</div>
        ) : (
          <PatientFeedingRecordTable
            rows={consolidatedPFRData || []}
            headers={patientFeedingRecordHeaders}
            useZebraStyles={true}
          />
        )}
      </div>
    </div>
  );
};

PatientFeedingRecord.propTypes = {};

export default PatientFeedingRecord;
