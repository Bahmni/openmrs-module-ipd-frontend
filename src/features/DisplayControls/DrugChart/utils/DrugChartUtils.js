import axios from "axios";
import moment from "moment";
import { MEDICATIONS_BASE_URL } from "../../../../constants";
import data from "../../../../utils/config.json";
import { AdminMedicationData } from "../components/AdminMedicationData";

const { config: { drugChart = {} } = {} } = data;

export const fetchMedications = async (patientUuid, forDate) => {
  const FETCH_MEDICATIONS_URL = `${MEDICATIONS_BASE_URL}?patientUuid=${patientUuid}&forDate=${forDate}`;
  try {
    const response = await axios.get(FETCH_MEDICATIONS_URL);
    return response;
  } catch (error) {
    console.error(error);
  }
};

const isLateTask = (startTime) => {
  const currentTime = Math.floor(new Date().getTime() / 1000);
  const lateTaskStatusWindowInSeconds =
    drugChart.timeInMinutesFromNowToShowPastTaskAsLate * 60;

  return startTime < currentTime - lateTaskStatusWindowInSeconds;
};

const isAdministeredLateTask = (startTime, effectiveStartDate) => {
  const lateTaskStatusWindowInSeconds =
    drugChart.timeInMinutesFromStartTimeToShowAdministeredTaskAsLate * 60;

  return effectiveStartDate - startTime > lateTaskStatusWindowInSeconds;
};

const checkIfSlotIsAdministered = (status) => {
  return status === "COMPLETED";
};

export const getUTCEpochForDate = (viewDate) => {
  const utcTimeEpoch = moment.utc(viewDate).unix();
  return utcTimeEpoch;
};

export const TransformDrugChartData = (drugChartData) => {
  const drugOrderData = [];
  const slotDataByOrder = [];
  console.log("drugChartData", drugChartData);

  AdminMedicationData.map((schedule) => {
    const { slots } = schedule;

    const administeredTimeInfo = [];

    slots.forEach((slot) => {
      const slotData = {};
      const { startTime, status, order, medicationAdministration } = slot;
      let medicationStatus = "Pending";
      let adminInfo = "",
        administeredTime;

      const isCompleted = checkIfSlotIsAdministered(status);

      if (isCompleted) {
        const isLate = isAdministeredLateTask(
          startTime,
          medicationAdministration.administeredDateTime
        );
        medicationStatus = isLate ? "Administered-Late" : "Administered";
        if (medicationAdministration) {
          const { administeredDateTime, provider } = medicationAdministration;
          const administeredDateTimeObject = new Date(
            administeredDateTime * 1000
          );
          administeredTime = moment(administeredDateTimeObject).format("HH:mm");
          adminInfo = provider.display + " [" + administeredTime + "]";
        } else {
          adminInfo = "";
        }
      }

      const drugOrder = {
        uuid: order.uuid,
        drugName: order.drug.display,
        drugRoute: order.route.display,
        administrationInfo: administeredTimeInfo,
        dosingInstructions: order.dosingInstructions,
      };

      if (order.duration) {
        drugOrder.duration = order.duration + " " + order.durationUnits.display;
      }
      if (order.doseUnits.display !== "ml") {
        drugOrder.dosage = order.dose;
        drugOrder.doseType = order.doseUnits.display;
      } else {
        drugOrder.dosage = order.dose + order.doseUnits.display;
      }
      if (order.duration) {
        drugOrder.duration = order.duration + " " + order.durationUnits.display;
      }

      const startDateTimeObj = new Date(startTime * 1000);

      const setLateStatus = isLateTask(startTime);
      const startHour = startDateTimeObj.getHours();
      const startMinutes = startDateTimeObj.getMinutes();
      slotData[startHour] = {
        minutes: startMinutes,
        status: !isCompleted && setLateStatus ? "Late" : medicationStatus,
        administrationInfo: adminInfo,
      };
      if (
        medicationStatus === "Administered" ||
        medicationStatus === "Administered-Late"
      ) {
        const adminData = {
          kind: medicationStatus,
          time: administeredTime,
        };
        drugOrder.administrationInfo.push(adminData);
      }
      if (
        !drugOrderData.some(
          (existingOrder) =>
            existingOrder.drugName === drugOrder.drugName &&
            existingOrder.uuid === drugOrder.uuid
        )
      ) {
        drugOrderData.push(drugOrder);
        slotDataByOrder.push(slotData);
      } else {
        const index = drugOrderData.findIndex(
          (existingOrder) =>
            existingOrder.drugName === drugOrder.drugName &&
            existingOrder.uuid === drugOrder.uuid
        );
        slotDataByOrder[index] = {
          ...slotDataByOrder[index],
          ...slotData,
        };
      }
    });
  });
  return [slotDataByOrder, drugOrderData];
};
