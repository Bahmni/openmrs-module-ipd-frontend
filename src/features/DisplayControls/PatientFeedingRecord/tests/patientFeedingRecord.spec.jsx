import React from "react";
import { getByText, render, waitFor } from "@testing-library/react";
import PatientFeedingRecordTable from "../components/PatientFeedingRecordTable";
import { patientFeedingRecordHeaders } from "../utils/PatientFeedingRecordUtils";
import PatientFeedingRecord from "../components/PatientFeedingRecord";
import { IPDContext } from "../../../../context/IPDContext";
import { mockConfig } from "../../../../utils/CommonUtils";
import MockDate from "mockdate";

const mockRowData = [
  {
    id: "1",
    dateAndTime: "2021-09-01T00:00:00.000Z",
    shift: "Morning",
    route: "Oral",
    feedType: "Milk",
    amount: "100 ml",
  },
  {
    id: "2",
    dateAndTime: "2021-09-01T00:00:00.000Z",
    shift: "Morning",
    route: "Oral",
    feedType: "Milk",
    amount: "100 ml",
  },
];

const mockVisitSummaryData = {
  uuid: "44832301-a09e-4bbb-b521-47144ed302cb",
  startDateTime: 1698301800000,
  stopDateTime: null,
  visitType: "IPD",
  admissionDetails: {
    uuid: "3a0642ef-2403-469d-8fd9-b2478a71e931",
    date: 1698301800000,
    provider: "Super Man",
    notes: null,
  },
  dischargeDetails: null,
};

describe("PatientFeedingRecordTable", () => {
  it("should render PatientFeedingRecordTable", () => {
    const { container } = render(
      <PatientFeedingRecordTable
        rows={mockRowData}
        headers={patientFeedingRecordHeaders}
        useZebraStyles={true}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should render rows with data", () => {
    const { getAllByTestId } = render(
      <PatientFeedingRecordTable
        rows={mockRowData}
        headers={patientFeedingRecordHeaders}
        useZebraStyles={true}
      />
    );
    expect(getAllByTestId("table-body-row").length).toBe(2);
  });
});

describe("PatientFeedingRecord", () => {
  it("should render PatientFeedingRecord", () => {
    const { container } = render(
      <IPDContext.Provider
        value={{ config: mockConfig, visitSummary: mockVisitSummaryData }}
      >
        <PatientFeedingRecord patientId="__test_patient_uuid__" />
      </IPDContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render current shift button", async () => {
    MockDate.set("2021-09-01T00:00:00.000Z");
    const { getByTestId, getByText } = render(
      <IPDContext.Provider
        value={{ config: mockConfig, visitSummary: mockVisitSummaryData }}
      >
        <PatientFeedingRecord patientId="__test_patient_uuid__" />
      </IPDContext.Provider>
    );
    const currentButton = getByTestId("current-period");
    expect(currentButton).toBeTruthy();
  });

  it("should render next shift button", async () => {
    MockDate.set("2021-09-01T00:00:00.00", async () => {
      MockDate.set("2021-09-01T00:00:00.000Z");
      const { getByTestId } = render(
        <IPDContext.Provider
          value={{ config: mockConfig, visitSummary: mockVisitSummaryData }}
        >
          <PatientFeedingRecord patientId="__test_patient_uuid__" />
        </IPDContext.Provider>
      );
      const nextButton = getByTestId("next-shift");
      expect(nextButton).toBeTruthy();
    });
  });
  it("should render previous shift button", async () => {
    MockDate.set("2021-09-01T00:00:00.00", async () => {
      MockDate.set("2021-09-01T00:00:00.000Z");
      const { getByTestId } = render(
        <IPDContext.Provider
          value={{ config: mockConfig, visitSummary: mockVisitSummaryData }}
        >
          <PatientFeedingRecord patientId="__test_patient_uuid__" />
        </IPDContext.Provider>
      );
      const previousButton = getByTestId("previous-shift");
      expect(previousButton).toBeTruthy();
    });
  });
  it("should show not in current shift message when not in current shift", async () => {
    MockDate.set("2021-09-01T00:00:00.00", async () => {
      MockDate.set("2021-09-01T00:00:00.000Z");
      const { getByTestId } = render(
        <IPDContext.Provider
          value={{ config: mockConfig, visitSummary: mockVisitSummaryData }}
        >
          <PatientFeedingRecord patientId="__test_patient_uuid__" />
        </IPDContext.Provider>
      );
      const previousButton = getByTestId("previous-shift");
      previousButton.click();
      await waitFor(() => {
        expect(getByText("You're not viewing the current period")).toBeTruthy();
      });
    });
  });
});
