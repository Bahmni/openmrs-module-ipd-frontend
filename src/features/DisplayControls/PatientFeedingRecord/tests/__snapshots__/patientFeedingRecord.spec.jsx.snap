// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientFeedingRecord should render PatientFeedingRecord 1`] = `
<div>
  <div
    class="container display-container"
  >
    <div
      class="navigation"
    >
      <div
        class="period-header"
      >
        <button
          class="margin-right-10 bx--btn bx--btn--tertiary bx--btn--expressive"
          data-testid="current-period"
          tabindex="0"
          type="button"
        >
          Current Period
        </button>
        <button
          class="margin-right-6 bx--btn bx--btn--tertiary bx--btn--expressive bx--tooltip--hidden bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--btn--icon-only--top bx--tooltip--align-center"
          data-testid="previous-shift"
          tabindex="0"
          type="button"
        >
          <div
            class="bx--assistive-text"
          />
          <svg
            aria-hidden="true"
            class="bx--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 8L10 3 10.7 3.7 6.4 8 10.7 12.3 10 13z"
            />
          </svg>
        </button>
        <button
          class="margin-right-10 bx--btn bx--btn--tertiary bx--btn--expressive bx--tooltip--hidden bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--btn--icon-only--top bx--tooltip--align-center"
          data-testid="next-shift"
          tabindex="0"
          type="button"
        >
          <div
            class="bx--assistive-text"
          />
          <svg
            aria-hidden="true"
            class="bx--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
            />
          </svg>
        </button>
        <div
          class="period-timing"
        >
          <div
            class="period-time"
          >
            31 Aug 2021
             
            <span
              class="mr-5"
            >
              |
            </span>
             
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16,30A14,14,0,1,1,30,16,14,14,0,0,1,16,30ZM16,4A12,12,0,1,0,28,16,12,12,0,0,0,16,4Z"
              />
              <path
                d="M20.59 22L15 16.41 15 7 17 7 17 15.58 22 20.59 20.59 22z"
              />
            </svg>
             
            08:00
             
            <span
              class="to-text"
            >
              to
            </span>
             
            01 Sep 2021
             
            <span
              class="mr-5"
            >
              |
            </span>
             
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16,30A14,14,0,1,1,30,16,14,14,0,0,1,16,30ZM16,4A12,12,0,1,0,28,16,12,12,0,0,0,16,4Z"
              />
              <path
                d="M20.59 22L15 16.41 15 7 17 7 17 15.58 22 20.59 20.59 22z"
              />
            </svg>
             
            08:00
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="no-pfr-data"
      >
        No patient feeding record data is available for the patient in this period
      </div>
    </div>
  </div>
</div>
`;

exports[`PatientFeedingRecordTable should render PatientFeedingRecordTable 1`] = `
<div>
  <div
    class="bx--data-table-content"
  >
    <table
      class="bx--data-table bx--data-table--normal bx--data-table--zebra bx--data-table--no-border"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            <div
              class="bx--table-header-label"
            >
              Date and Time
            </div>
          </th>
          <th
            scope="col"
          >
            <div
              class="bx--table-header-label"
            >
              Shift
            </div>
          </th>
          <th
            scope="col"
          >
            <div
              class="bx--table-header-label"
            >
              Route
            </div>
          </th>
          <th
            scope="col"
          >
            <div
              class="bx--table-header-label"
            >
              Feed Type
            </div>
          </th>
          <th
            scope="col"
          >
            <div
              class="bx--table-header-label"
            >
              Amount
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        aria-live="polite"
      >
        <tr
          data-testid="table-body-row"
        >
          <td>
            2021-09-01T00:00:00.000Z
          </td>
          <td>
            Morning
          </td>
          <td>
            Oral
          </td>
          <td>
            Milk
          </td>
          <td>
            100 ml
          </td>
        </tr>
        <tr
          data-testid="table-body-row"
        >
          <td>
            2021-09-01T00:00:00.000Z
          </td>
          <td>
            Morning
          </td>
          <td>
            Oral
          </td>
          <td>
            Milk
          </td>
          <td>
            100 ml
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
